integrations/PinterestTag/browser.js

```diff
@@ -165,42 +160,51 @@ export default class PinterestTag {
         // This check fails if user is sending boolean value as false
         // Adding toString because if the property value is boolean then it never gets reflected in destination
         if (isDefinedAndNotNull(flattenPayload[custom.properties])) {
-          pinterestObject[custom.properties] =
-            flattenPayload[custom.properties].toString();
+          pinterestObject[custom.properties] = flattenPayload[custom.properties].toString();
         }
       });
     }
     return pinterestObject;
   }
 
-  /**
-   * This gives destination events .
-   * Logics: If our eventMapping is not able to map the event that is sent by user payload then it will look into
-   * userDefinedEventsMapping array. In case if it is not found there as well, it will return "custom".
-   * @param {rudder event name} event
-   * @returns
-   */
   getDestinationEventName(event) {
+    let eventNames;
+    /*
+    Step 1: At first we will look for
+            the event mapping in the UI. In case it is similar, will map to that.
+     */
     if (this.userDefinedEventsMapping.length > 0) {
-      const userDefinedEvent = this.userDefinedEventsMapping.find(
-        (e) => e.from.toLowerCase() === event.toLowerCase()
+      const keyMap = getHashFromArrayWithDuplicate(
+        this.userDefinedEventsMapping,
+        'from',
+        'to',
+        false
       );
-      if (userDefinedEvent && userDefinedEvent.to) {
-        return {
-          dest: userDefinedEvent.to,
-          isUserDefinedEvent: true,
-        };
-      }
+      eventNames = keyMap[event];
     }
-    const destinationEvent = eventMapping.find((p) =>
-      p.src.includes(event.toLowerCase())
-    );
-    if (!destinationEvent) {
-      return {
-        dest: "custom",
-      };
+    if (isDefined(eventNames)) {
+      return eventNames;
     }
-    return destinationEvent;
+    /*
+    Step 2: To find if the particular event is amongst the list of standard
+            Rudderstack ecommerce events, used specifically for Pinterest Conversion API
+            mappings.
+    */
+    if (!eventNames) {
```

Comment:
This check is not required. We are already returning if `eventNames` is defined.

Reply:
addressed